lib/srv/regular/sshserver.go

```diff
@@ -1470,9 +1470,15 @@ func (s *Server) handleSessionRequests(ctx context.Context, ccx *sshutils.Connec
 			)
 
 			if err := s.dispatch(ctx, ch, req, scx); err != nil {
-				s.replyError(ch, req, err)
-				span.End()
-				return
+				// a trace.NotImplemented error here indicates that a reply to the request was already handled inside
+				// the dispatch method, so we set the wantReply flag to false to avoid sending a second reply below.
+				if trace.IsNotImplemented(err) {
+					req.WantReply = false
+				} else {
```

Comment:
This is kinda nasty. I should probably refactor the `dispatch` method to return `(bool, error)` instead and use that to indicate whether the reply has already been processed.